channel/state/outcome/guarantee.go

```diff
@@ -0,0 +1,60 @@
+package outcome
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// A Guarantee is an Allocation with AllocationType == GuaranteeAllocationType and Metadata = encode(GuaranteeMetaData)
+type GuaranteeMetadata struct {
```

Comment:
The `Left`, `Right` guarantee is a special case of the prior, more general ordered list. SATP virtual funding uses this special case, but is it better for the package to retain the more flexible format?

My instinct: probably not, as the larger list serviced the prior recursive VC construction, and (as far as I know) no other purpose.